src/server/editorServices.ts

```diff
@@ -3163,32 +3208,37 @@ namespace ts.server {
             // Work on array copy as we could add more projects as part of callback
             for (const project of arrayFrom(this.configuredProjects.values())) {
                 // If this project has potential project reference for any of the project we are loading ancestor tree for
-                // we need to load this project tree
-                if (forEachPotentialProjectReference(
-                    project,
-                    potentialRefPath => forProjects!.has(potentialRefPath)
-                ) || forEachResolvedProjectReference(
-                    project,
-                    (_ref, resolvedPath) => forProjects!.has(resolvedPath)
-                )) {
-                    // Load children
-                    this.ensureProjectChildren(project, seenProjects);
+                // load this project first
+                if (forEachPotentialProjectReference(project, potentialRefPath => forProjects!.has(potentialRefPath))) {
+                    updateProjectIfDirty(project);
                 }
+                this.ensureProjectChildren(project, forProjects, seenProjects);
             }
         }
 
-        private ensureProjectChildren(project: ConfiguredProject, seenProjects: Set<NormalizedPath>) {
+        private ensureProjectChildren(project: ConfiguredProject, forProjects: ReadonlyCollection<string>, seenProjects: Set<NormalizedPath>) {

```

Comment:
It's a little unfortunate that "project" refers to a `ConfiguredProject` in the first parameter, a `string` in the second, and a `NormalizedPath` in the third.

Reply:
forProjects can contain not configured projects and hence its not NormalizedPath. this just avoids having to filter out non configured project by doing this. In practice most of the time its going to be configured project itself but string represents correct type for `forProjects` seenProjects is always going to be configured project hence the type for the key is correct.

Reply:
Sorry, I didn't mean that I thought there was a bug - I meant that variable names frequently convey something about the type of objects they're expected to contain and that one might, therefore, expect these three parameters to have similar types.

One way to address this (which isn't necessary - this is just an example) would be to rename "forProjects" to "forProjectPaths", etc.